src/lightning/pytorch/trainer/connectors/logger_connector/result.py

```diff
@@ -265,7 +265,7 @@ def _wrap_compute(self, compute: Any) -> Any:
         # Override to avoid syncing - we handle it ourselves.
         @wraps(compute)
         def wrapped_func(*args: Any, **kwargs: Any) -> Optional[Any]:
-            if not self._update_called:
+            if not self.update_called:
```

Comment:
The public property [was introduced in in torchmetrics 1.0.0](https://github.com/Lightning-AI/torchmetrics/commit/6e2213a65c6bbd0af975bb487759dfd009137112), but Lightning supports[ >=0.7.0](https://github.com/Lightning-AI/lightning/blob/343f80436d3233bac155e77b92891f6281ff90d1/requirements/pytorch/base.txt#L9). So there are two options I see:
A) Bump the minimum torchmetrics requirement to 1.0.0
B) Add a switch here that >= 1.0.0 will use the public attribute, while < 1.0.0 still uses the protected one.

Reply:
@matsumotosan this way you can use a switch for the torchmetrics version as suggested in B.

```suggestion
            tm_version = packaging.version.parse(torchmetrics.__version__)
            update_called_attribute = self.update_called if tm_version >= packaging.version.parse("1.0.0") else self._update_called
            if not update_called_attribute:
```